### PR TITLE
[Draft][onert-micro / circle-execution-plan] Enable static memory manager for onert-micro

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This PR enables circle-execution-plan and static memory manager for onert-micro.

for issue: #9849 

ONE-DCO-1.0-Signed-off-by: Artem Balyshev a.balyshev@partner.samsung.com

### CODE REVIEW

File: compiler/circle-execution-plan/src/CircleExecutionPlan.cpp [hunk 1]
```diff
@@ -36,6 +36,29 @@ int entry(int argc, char **argv)
   arser.add_argument("input").help("Input circle model");
   arser.add_argument("output").help("Output circle model");
   arser.add_argument("--platform").default_value("linux").help("Platform name: linux mcu cmsisnn");
+  arser.add_argument("--buffer_type")
```
Comment (binarman) [inline compiler/circle-execution-plan/src/CircleExecutionPlan.cpp:39]: Maybe name it like "--allocating_mode" or "--allocating_strategy"?

File: compiler/circle-execution-plan/src/ExecutionPlanner.cpp [hunk 2]
```diff
@@ -345,7 +554,7 @@ void ExecutionPlanner::create_alloc_node_inform_vector(bool null_consts, bool nu
   for (size_t i = 0; i < _ordered_nodes.size(); i++)
   {
     auto circle_node = loco::must_cast<luci::CircleNode *>(_ordered_nodes[i]);
-    auto node_size = 1;
+    auto node_size = circle_node->rank() > 0 ? 1 : 0;
```
Comment (binarman) [inline compiler/circle-execution-plan/src/ExecutionPlanner.cpp:557]: Are you sure?
node with rank == 0 is a scalar

File: compiler/circle-execution-plan/src/ExecutionPlanner.cpp [hunk 3]
```diff
@@ -86,18 +86,195 @@ void create_allocation_node(Json::Value &allocations_node,
...
+  const auto input_required_size = get_offsets_with_greedy_by_size();
+
+  for (uint32_t i = 0; i < _ordered_nodes.size(); i++)
+  {
+    luci::CircleNodeExecutionPlan execution_plan(i, _offsets[i]);
+    luci::add_execution_plan(loco::must_cast<luci::CircleNode *>(_ordered_nodes[i]),
+                             execution_plan);
+  }
+  dump_inform();
+  printf("Input graph buffer required memory = %d\n", input_required_size);
```
Comment (binarman) [inline compiler/circle-execution-plan/src/ExecutionPlanner.cpp:152]: Maybe move this (and output/intermediate tensor) code to a separate function?

File: compiler/circle-execution-plan/src/ExecutionPlanner.cpp (hunk 2 above)
Comment (BalyshevArtem) [reply compiler/circle-execution-plan/src/ExecutionPlanner.cpp:557]: done, please see https://github.com/Samsung/ONE/pull/9893

File: compiler/circle-execution-plan/src/ExecutionPlanner.cpp (hunk 3 above)
Comment (BalyshevArtem) [reply compiler/circle-execution-plan/src/ExecutionPlanner.cpp:152]: done, please see https://github.com/Samsung/ONE/pull/9893

**Comment by BalyshevArtem** [comment]
everything is merged, close it

**Pull request closed without merging**